src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -95,7 +95,7 @@ public function setChanges(array $changes)
     /**
      * Sets a factory.
      *
-     * @param string|array $factory A PHP function or an array containing a class/Reference and a method to call
+     * @param string|array|Reference $factory A PHP callable or an array containing a class/Reference and a method to call
```

Comment:
`or a reference` in the description then, to describe the new choice.

And this `function` to `callable` should be reverted. The array being passed is not a callable (we don't accept `[$object, $method]` as we cannot dump it; a reference has to be used)